engine/h2shared/vid_cgx.c

```diff
@@ -487,6 +486,10 @@ static void VID_NumModes_f (void)
 
 static void VID_DestroyWindow (void)
 {
+	extern cvar_t v_gamma;
+	// always call VID_ShiftPalette after mode changes
```

Comment:
I don't get this. Are we not always calling VID_SetPalette already, in VID_SetMode?

Reply:
Unfortunately that only sets the base palette before gamma correction. By flagging v_gamma as changed there will also be call to VID_ShiftPalette, which sets the final gamma corrected palette.

Reply:
Ah, because you removed vid_curpal and calling SetMode with NULL palette.

What happens when you first run the game - are you still getting the same brightness
saved in config.cfg?

How much performance difference is this really making for us?

Reply:
Palette updates in native modes have always been sluggish, so I tried to remove everything non-essential from VID_SetPalette. I found that backing up every palette update with vid_curpal is not necessary if I force a gamma update after the mode change. This way I always get the palette with the correct brightness.